src/CarbonAware/src/Interfaces/ICarbonIntensityDataSource.cs

```diff
@@ -22,7 +22,7 @@ public interface ICarbonIntensityDataSource
     /// <param name="periodStartTime">The start time of the period.</param>
     /// <param name="periodEndTime">The end time of the period.</param>
     /// <returns>A list of emissions data for the given time period.</returns>
-    public Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(IEnumerable<Location> locations, DateTimeOffset periodStartTime, DateTimeOffset periodEndTime);
+    public Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(IEnumerable<Location> locations, DateTimeOffset periodStartTime, DateTimeOffset periodEndTime, bool bestValue = false);
```

Comment:
Can you explain more about this interface change?  Looks like you don't utilize this new parameter anywhere.  I might take a YAGNI approach here and wait until we plan on using it to wire up such a change.

Reply:
am ok calling this change 'YAGNI' for today, and us removing it, since the fix is limited to the aggregator -- and defer to later whether the dataSources own determining 'best'

Reply:
Happy to jump on a call and explain it, but Dan and I basically talked out what each layer of the code is supposed to be responsible for and this was the solution that "felt" best to me.